public/content/en/gems/contract-programming.md

```diff
@@ -96,17 +109,34 @@ struct Date {
             &this.month,
             &this.day);
     }
+
+    // Serializes Date object to YYYY-MM-DD
+	string toString() const
+    out (result) {
+        import std.algorithm: all, count, equal, map;
+        import std.string: isNumeric;
+        import std.array: split;
+
+		// verify we return YYYY-MM-DD
+        assert(result.count("-") == 2);
+        auto parts = result.split("-");
+        assert(parts.map!`a.length`.equal([4, 2, 2]));
+        assert(parts.all!isNumeric);
+    }
```

Comment:
Was an issue with tabs in the markdown file - maybe we should forbid them completely there?